osu.Game/Screens/Play/Player.cs

```diff
@@ -297,11 +297,17 @@ private void load(AudioManager audio, OsuConfigManager config, OsuGameBase game)
             ScoreProcessor.HasCompleted.BindValueChanged(scoreCompletionChanged);
             HealthProcessor.Failed += onFail;
 
-            foreach (var mod in Mods.Value.OfType<IApplicableToScoreProcessor>())
-                mod.ApplyToScoreProcessor(ScoreProcessor);
+            ScoreProcessor.OnLoadComplete += _ =>
```

Comment:
We tend to avoid using `OnLoadComplete` whenever possible. Have you tried instead moving this code to `LoadComplete` of `Player`?

Reply:
Oh, hmm... will give that a try.

Reply:
One way without using `OnLoadComplete` is using `ScheduleAfterChildren`, to apply after the judgement processors have been fully loaded, though I'm still not sure what's exactly causing the issue, as applying them on `Player.LoadComplete` or `OnEntering` still doesn't work and leaves a stale transform with an incorrect clock start time.

Reply:
At *very least*, if we are going to do things the way you have it, there needs to be an inline comment explaining this.